runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java

```diff
@@ -406,4 +415,23 @@ private boolean slotContainsText(UIActionSlot slot, String text)
 		return slot.getAction().getName().toLowerCase().contains(text.toLowerCase());
 	}
 
+	/**
+	 * Updates the current skill calculator (if present) by looking up
+	 * the player's current XP and level for that skill
+	 * <p>
+	 * This method is invoked at a defined internal,
+	 * see: {@link SkillCalculatorPlugin#updateSkillCalculator()}
+	 */
+	void updateSkillCalculator()
+	{
+		// If the user has selected a calculator, update its fields
+		Optional.ofNullable(calculatorType).ifPresent(calc ->
```

Comment:
this is just `if (calc != null) {`

Reply:
I liked the readability of ifPresent

Reply:
Still it should be changed to null check, because you are not doing anything with the Optional after

Reply:
Done